src/core/ccitt_stream.js

```diff
@@ -0,0 +1,60 @@
+/* Copyright 2012 Mozilla Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Dict, isDict } from './primitives';
+import { CCITTFaxDecoder } from './ccitt';
+import { DecodeStream } from './stream';
+
+var CCITTFaxStream = (function CCITTFaxStreamClosure() {
+  function CCITTFaxStream(str, maybeLength, params) {
+    this.str = str;
+    this.dict = str.dict;
+
+    if (!isDict(params)) {
+      params = Dict.empty;
+    }
+    this.ccittFaxDecoder = new CCITTFaxDecoder(str, {
```

Comment:
I was thinking we could wrap `str` into `source` above, e.g. `const source = { next() { return str.getByte(); } };`, but this will work fine for now. Next CCITTFaxDecoder user need to follow the same interface of `{ getByte() { ... }}`

Reply:
> I was thinking we could wrap str into source above, e.g. const source = { next() { return str.getByte(); } };, but this will work fine for now.

That seems like a nice idea, so I figured that it cannot hurt to implement that while we're at it.